frappe/model/base_document.py

```diff
@@ -375,12 +375,21 @@ def get_parentfield_of_doctype(self, doctype):
 		fieldname = [df.fieldname for df in self.meta.get_table_fields() if df.options==doctype]
 		return fieldname[0] if fieldname else None
 
-	def db_insert(self):
-		"""INSERT the document (with valid columns) in the database."""
+	def db_insert(self, pg_ignore_pk_violation=False):
```

Comment:
Shouldn't this be DB agnostic?

```suggestion
	def db_insert(self, ignore_if_duplicate=False):
```

On another note, I feel UX across DB types should be consistent (wrt Frappe). So, if MariaDB doesn't cause a ruckus in case of a constraint violation, Postgres also shouldn't.

Reply:
It can't be made DB agnostic. hence added `pg` in arg name. Check note about mariadb in post, there's no equivalent method that JUST ignores `name` collision without doing more.

Reply:
We can do same thing that's happening in `Document.insert` catch and ignore primary key violations 🤔